kratos/utilities/multipoint_constraint.h

```diff
@@ -0,0 +1,611 @@
+            NodeType &node = rNodes[slave_node_id];
+            Node<3>::DofsContainerType::iterator it = node.GetDofs().find(slave_dof_key);
+            constraint_eq_data->SetSlaveEquationId(it->EquationId());
+            for (auto &master_data : *constraint_eq_data)
+            {
+                size_t master_dof_key = master_data->MasterDofKey();
+                size_t master_dof_id = master_data->MasterDofId();
+
+                NodeType &master_node = rNodes[master_dof_id];
+                Node<3>::DofsContainerType::iterator it_master = master_node.GetDofs().find(master_dof_key);
+                master_data->SetMasterEqId(it_master->EquationId());
+            }
+        } 
+    }
+
+    virtual void ExecuteAfterSolving(TSystemMatrixType &rA,
+                                     TSystemVectorType &rDx,
+                                     TSystemVectorType &rb) override
+    {
+        for (auto &constraint_eq_data : this->GetData())
+        {
+            size_t slave_equation_id = constraint_eq_data->SlaveEquationId();
+            for (auto &master_data : *constraint_eq_data)
+            {
+                rDx[slave_equation_id] = TSparseSpace::GetValue(rDx, slave_equation_id) + TSparseSpace::GetValue(rDx, master_data->MasterEqId()) * master_data->MasterWeight();
+            }
+
+            rDx[slave_equation_id] = TSparseSpace::GetValue(rDx, slave_equation_id) + constraint_eq_data->ConstantUpdate();
+            constraint_eq_data->SetConstantUpdate(0.0);
+        }        
+    }
+
+    virtual void ModifyEquationIdsForConstraints(Element &rCurrentElement,
+                                                         EquationIdVectorType &rEquationId,
+                                                         ProcessInfo &rCurrentProcessInfo) override
+    {
+        const size_t number_of_nodes = rCurrentElement.GetGeometry().PointsNumber();
+        // For each node check if it is a slave or not If it is .. we change the Transformation matrix
+        for (size_t j = 0; j < number_of_nodes; j++)
+        {            
+            DofsVectorType element_dofs;
+            rCurrentElement.GetDofList(element_dofs, rCurrentProcessInfo);
+            int num_dofs_per_node = element_dofs.size() / number_of_nodes;
+            if (rCurrentElement.GetGeometry()[j].Is(SLAVE))
+            { //temporary, will be checked once at the beginning only
+                // Necessary data for iterating and modifying the matrix
+                int start_position_node_dofs = num_dofs_per_node * (j);
+                for (int i = 0; i < num_dofs_per_node; i++)
+                {
+                    if (this->GetData().GetNumberOfMasterDofsForSlave(*(element_dofs[start_position_node_dofs + i])) > 0)
+                    {
+                        auto &constraint_eq_data = this->GetData().GetConstraintEquation(*(element_dofs[start_position_node_dofs + i]));
+                        for (auto &master_data : constraint_eq_data)

```

Comment:
const?

Reply:
The iterator here itself is a const .. so const here is not necessary.